### PR TITLE
Rework Plot Axes

### PR DESCRIPTION
**Pull request by johannes-wolf**
Some cleanup of the axis drawing code. More refactoring should follow, but in a follow-up PR.

Axis styling documentation is not good yet!
New features:
- Per axis coloring for scientific plots
- Axis padding for both scientific and school-book plots
- Axis "overshoot" for school-book plots
- Customizable axis, tick and grid z-layer
- One can disable the shared "0" tick

I am not happy with the shared "0" tick drawing, this should get improved!

### CODE REVIEW

File: src/process.typ [hunk 1]
```diff
@@ -56,14 +56,16 @@
   let drawables = ()
   let bounds = none
 
-  for el in body {
-    let r = element(ctx, el)
-    if r != none {
-      if r.bounds != none {
-        bounds = aabb.aabb(r.bounds, init: bounds)
+  if body != none {
```
Comment (fenjalien) [inline src/process.typ:59]: Why not panic? Why pass silently?

File: tests/chart/ref.png
Comment (fenjalien) [inline tests/chart/ref.png:1]: Is the padding on the second chart okay?

**Review by fenjalien — CHANGES_REQUESTED** [review]
Are you happy with the state of `axes.typ`? I haven't focused on it as I don't mind but I can take a closer look/be more picky if you want.

Are you going to fix the docs in this PR?

UPdate the changelog!

File: src/lib/axes.typ [hunk 2]
```diff
@@ -3,29 +3,66 @@
...
 #import "/src/vector.typ"
 #import "/src/styles.typ"
+#import "/src/process.typ"
+#import "/src/drawable.typ"
+#import "/src/path-util.typ"
 
 #let typst-content = content
 
-// Global defaults
+/// Default axis style
```
[the remark this answers is not in the document]
Comment (johannes-wolf) [reply src/lib/axes.typ:12]: Yes.

File: src/process.typ (hunk 1 above)
Comment (johannes-wolf) [reply src/process.typ:59]: An empty (group) body should be legal. Example:
```typst
group({
  if ... {
    draw(...)
  }
})
```
Has the body set to `none` if the if-condition is false. This should not panic, but still create a group (or any other grouping element).

**Comment by johannes-wolf** [comment]
> he docs in this PR?

No, I am not yet happy but would like to do further changes later.
Docs for styling are now added (although they are quite minimal).

File: src/process.typ (hunk 1 above)
Comment (fenjalien) [reply src/process.typ:59]: I would prefer the check to be in group then instead of `process.many` which should always expect an array

Comment (johannes-wolf) [reply src/process.typ:59]: But isn't `none` or `()` like kind of the same? I mean, the ctx function could also return `none` which would be a second check… I favour having this at the lower level, I would even propose allowing passing functions to `process.element` and `*.many`.

Comment (johannes-wolf) [reply src/process.typ]: Should be solved with #491.

**Review by fenjalien — APPROVED** [review]
Make sure to rebase this first, as `resolve-body` is still in `grouping.typ`

**Pull request merged**